docs/pages/build/advanced-credentials-configuration.md

```diff
@@ -0,0 +1,262 @@
+
+```
+ios/certs/*
+```
+
+### Configuring credentials.json
+
+Create (or edit) `credentials.json` and configure it with the credentials:
+
+```json
+{
+  "android": {
+    ...
+  },
+  "ios": {
+    "provisioningProfilePath": "ios/certs/profile.mobileprovision",
+    "distributionCertificate": {
+      "path": "ios/certs/dist.p12",
+      "password": "DISTRIBUTION_CERTIFICATE_PASSWORD"
+    }
+  }
+}
+```
+
+- `provisioningProfilePath` points to the Provisioning Profile located on your computer. Both relative (to the project root) and absolute paths are supported.
+- `distributionCertificate.path` points to the Distribution Certificate located on your computer. Both relative (to the project root) and absolute paths are supported.
+- `distributionCertificate.password` is the password for the Distribution Certificate located at `distributionCertificate.path`.
+
+## Setting Credentials Source in eas.json
+
+### Auto Mode (Default)
+
+Let's assume we're only building for Android and we're using the following configuration (defined in `eas.json` - [learn more about this file](../eas-json/)):
+
+```json
+{
+  "builds": {
+    "android": {
+      "release": {
+        "workflow": "generic"
+      }
+    }
+  }
+}
+```
+
+Given that configuration, `expo eas:build` resolves the credentials in the so-called **auto mode**.
+
+The algorithm of the auto mode works like this:
+
+- If the entry for a given platform in `credentials.json` is defined and the project's credentials exist on the Expo servers:
+  - Check if they are the same and if so - use them.
+  - Otherwise, display a prompt to ask which ones should be used.
+- If the entry for a given platform in `credentials.json` is defined but the project's credentials do **not** exist on the Expo servers - use the credentials defined in `credentials.json`.
+- If the entry for a given platform in `credentials.json` is **not** defined but the project's credentials exist on the Expo servers - use the credentials from the Expo servers.
+- If neither the entry for a given platform in `credentials.json` is defined nor remote credentials exist - display a prompt to ask whether new credentials should be generated and stored on the Expo servers.
+
+### Local Mode
+
+You can configure EAS Builds so that `expo eas:build` will be reading the credentials only from `credentials.json`. Just set `"credentialsSource": "local"` in one of your [build profiles](../eas-json/) in `eas.json`.
+
+Example:
+
+```json
+{
+  "builds": {
+    "android": {
+      "release": {
+        "workflow": "generic",
+        "credentialsSource": "local"
+      }
+    }
+  }
+}
+```
+
+This can be particularly useful when running the build on a CI. If the `credentialsSource` is set to `local` but the `credentials.json` file doesn't exist Expo CLI will throw an error.
```

Comment:
```suggestion
This can be particularly useful when running the build on CI. If the `credentialsSource` is set to `local` but the `credentials.json` file doesn't exist Expo CLI will throw an error.
```